pkg/stub/secrets.go

```diff
@@ -35,22 +35,25 @@ func (h *Handler) copyDefaultClusterPullSecret() error {
 	secretToCreate.Annotations = make(map[string]string)
 	secretToCreate.Annotations[v1.SamplesVersionAnnotation] = v1.GitVersionString()
 	_, err = h.secretclientwrapper.Create("openshift", &secretToCreate)
-	return err
+	if kerrors.IsAlreadyExists(err) {
+		h.secretclientwrapper.Update("openshift", &secretToCreate)
+	}
+
 }
 
 func (h *Handler) manageDockerCfgSecret(deleted bool, Config *v1.Config, s *corev1.Secret) error {
 	secret := s
 	var err error
 	if secret == nil {
-		secret, err = h.secretclientwrapper.Get(h.namespace, v1.SamplesRegistryCredentials)
+		secret, err = h.secretclientwrapper.Get(coreosPullSecretNamespace, coreosPullSecretName)
 		if err != nil && kerrors.IsNotFound(err) {
 			return nil
 		}
 		if err != nil {
 			return err
 		}
 	}
-	if secret.Name != v1.SamplesRegistryCredentials {
+	if secret.Name != coreosPullSecretName && secret.Name != v1.SamplesRegistryCredentials {
```

Comment:
shouldn't this only be the coreosPullSecretName ?

Reply:
my thought was if they delete the secret in the openshift namespace, we want to recreate, no?

that said, after putting that marker, I forgot to tweak the delete processing below to do that

i'll update and push

Reply:
ok.  if this function really can do something intelligent with both the kube-system secret and the openshift secret, this makes more sense (though you should probably be checking the namespace too, then...  otherwise this function risks taking action on a "coreos pull secret" that exists in the "openshift" namespace, for example.

Reply:
actually I forgot the recreate on delete is handled elsewhere

sorting through some of the flows now with the unit tests

it may in fact turn out we only process coreosPullSecretName in this path

will clarify on repush